lib/_http_common.js

```diff
@@ -247,44 +246,44 @@ exports.httpSocketSetup = httpSocketSetup;
  * so take care when making changes to the implementation so that the source
  * code size does not exceed v8's default max_inlined_source_size setting.
  **/
-function isValidTokenChar(ch) {
-  if (ch >= 94 && ch <= 122)
-    return true;
-  if (ch >= 65 && ch <= 90)
-    return true;
-  if (ch === 45)
-    return true;
-  if (ch >= 48 && ch <= 57)
-    return true;
-  if (ch === 34 || ch === 40 || ch === 41 || ch === 44)
+var validTokens = [
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // 0 - 15
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // 16 - 31
+  0, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1, 1, 0, // 32 - 47
+  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, // 48 - 63
+  0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, // 64 - 79
+  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1, // 80 - 95
+  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, // 96 - 111
+  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 0, // 112 - 127
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // 128 ...
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0  // ... 255
+];
+function checkIsHttpToken(val) {
+  if (typeof val !== 'string' || val.length === 0)
+    return false;
+  if (!validTokens[val.charCodeAt(0)])
     return false;
-  if (ch >= 33 && ch <= 46)
+  if (val.length < 2)
     return true;
-  if (ch === 124 || ch === 126)
+  if (!validTokens[val.charCodeAt(1)])
+    return false;
+  if (val.length < 3)
     return true;
```

Comment:
you might want to check if setting a value and having a single return improves things.